crates/ruff_linter/resources/test/fixtures/flake8_return/RET501.py

```diff
@@ -17,3 +17,28 @@ def get(self, key: str) -> None:
     def prop(self) -> None:
         print("Property not found")
         return None
+
+
+import abc
+import enum
+import types
+from functools import cached_property
+
+
+class BaseCache2:
+    @cached_property
+    def prop(self) -> None:
+        print("Property not found")
+        return None
+
+    @abc.abstractproperty
+    def prop2(self) -> None:
+        return None
```

Comment:
Note: single line functions are ignored anyway.
So the print is needed for sensible testing.

Reply:
Ah, thank you!